resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/features/BasicDriverFeatureStepSuite.scala

```diff
@@ -87,6 +87,10 @@ class BasicDriverFeatureStepSuite extends SparkFunSuite {
     assert(configuredPod.container.getImage === "spark-driver:latest")
     assert(configuredPod.container.getImagePullPolicy === CONTAINER_IMAGE_PULL_POLICY)
 
+    val expectedPortNames = Set(DRIVER_PORT_NAME, BLOCK_MANAGER_PORT_NAME, UI_PORT_NAME)
+    val foundPortNames = configuredPod.container.getPorts.asScala.map(port => port.getName).toSet
```

Comment:
Better to be safer here and check the protocol and the target port numbers. You can just compare against a list of expected `ContainerPort` objects - Kubernetes objects in our Java API have been good about giving us sane `equals` implementations.